### PR TITLE
add syntax sugar for merging two trajs

### PR DESCRIPTION
**Pull request by hainm**
```traj = traj0 + traj1```

### CODE REVIEW

File: pytraj/trajectory.py [hunk 1]
```diff
@@ -947,3 +947,44 @@ def _apply(self, func):
...
+        >>> traj1.xyz[0, 0] == traj3.xyz[0, 0]
+        array([ True,  True,  True], dtype=bool)
+        >>> traj2.xyz[0, -1] == traj3.xyz[0, -1]
+        array([ True,  True,  True], dtype=bool)
+
+        See also
+        --------
+        pytraj.tools.merge_trajs
+
+            - this method is more memory efficient
```
Comment (swails) [inline pytraj/trajectory.py]: Then why not implement it using that instead?

**Comment by swails** [comment]
I have to say when I saw the commit log, I expected extending trajectories along the frame dimension, *not* the atom dimension.  Similar to `list.__add__`.  But we probably think about Trajectory math differently.

File: pytraj/trajectory.py (hunk 1 above)
Comment (hainm) [inline pytraj/trajectory.py]: good point. Actually I copied from `that`. Will delete the comment.

**Comment by hainm** [comment]
> I have to say when I saw the commit log, I expected extending trajectories along the frame dimension, not the atom dimension. 

there is no official behaviour yet. I just try to follow `parmed` syntax to join/merge two parms together. 

anyway, `pytraj.Trajectory` has `append` method so users can append new Frame to it.

**Pull request merged**